collect_app/src/main/java/org/odk/collect/android/preferences/DisabledPreferencesRemover.java

```diff
@@ -0,0 +1,79 @@
+package org.odk.collect.android.preferences;
+
+import android.content.SharedPreferences;
+import android.preference.Preference;
+import android.preference.PreferenceCategory;
+import android.preference.PreferenceScreen;
+import android.util.Log;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+
+import static org.odk.collect.android.preferences.PreferencesActivity.INTENT_KEY_ADMIN_MODE;
+
+class DisabledPreferencesRemover {
+    private final String TAG = getClass().getSimpleName();
```

Comment:
Unfortunately, I don't think we can use this if we're using Proguard for obfuscation.